src/pinning.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "mamba/pinning.hpp"
+
+#include <fstream>
+
+
+namespace mamba
+{
+    void pin_python_spec(const PrefixData& prefix_data, std::vector<std::string>& specs)
+    {
+        bool py_found = false;
+        std::string py_spec = "";
+
+        for (auto spec : specs)
+        {
+            if (spec.find("python") != std::string::npos)
```

Comment:
I am not sure this works. Can we not use the MatchSpec facilities for proper parsing?

Reply:
fixed!